### PR TITLE
Rover: Added reversing loiter_mode

### PR DESCRIPTION
**Pull request by sahith22**
I have added parameter 2 to loiter type. This allows the boat to face its stern towards the target and move towards it.

### CODE REVIEW

**Comment by IamPete1** [comment]
@sahith22 Are you happy for me to squash your commits for you?

File: APMrover2/mode_loiter.cpp [hunk 1]
```diff
@@ -43,13 +43,13 @@ void ModeLoiter::update()
        // calculate bearing to destination
         _desired_yaw_cd = rover.current_loc.get_bearing_to(_destination);
         float yaw_error_cd = wrap_180_cd(_desired_yaw_cd - ahrs.yaw_sensor);
-        // if destination is behind vehicle, reverse towards it
-        if (fabsf(yaw_error_cd) > 9000 && g2.loit_type == 0) {
+        // if destination is behind vehicle, reverse towards it
```
Comment (amilcarlucas) [inline APMrover2/mode_loiter.cpp]: Whitespace at end of line

File: APMrover2/mode_loiter.cpp [hunk 2]
```diff
@@ -43,13 +43,13 @@ void ModeLoiter::update()
...
-        // if destination is behind vehicle, reverse towards it
-        if (fabsf(yaw_error_cd) > 9000 && g2.loit_type == 0) {
+        // if destination is behind vehicle, reverse towards it    
+        if ((fabsf(yaw_error_cd) > 9000 && g2.loit_type == 0) || g2.loit_type == 2){
             _desired_yaw_cd = wrap_180_cd(_desired_yaw_cd + 18000);
             yaw_error_cd = wrap_180_cd(_desired_yaw_cd - ahrs.yaw_sensor);
             _desired_speed = -_desired_speed;
         }
-
+
```
Comment (amilcarlucas) [inline APMrover2/mode_loiter.cpp]: Unnecessary whitespace change

**Comment by amilcarlucas** [comment]
It can now be rebased and squashed.

**Comment by IamPete1** [comment]
I manged to break this while attempting to squash, Sorry! This git stuff is tricky, I have [re-opened](https://github.com/ArduPilot/ardupilot/pull/12742) it from my backup copy of this branch. I think once I had updated this to master there was no difference so the PR was closed then I was no longer allowed to push the squashed commit?

**Comment by sahith22** [comment]
The master does not contain the updated code.

**Comment by IamPete1** [comment]
maybe I should add that you can fix it with
git remote add iampete1 https://github.com/IamPete1/ardupilot.git
git fetch iampete1
git cherry-pick ed6a47e45ed54e8569f0916663b87eb4f725a59a
git push -f

**Comment by sahith22** [comment]
upon using git push-f , I am told that it is unable to access 'https://github.com/IamPete1/ardupilot.git/': The requested URL returned error: 403.

**Comment by IamPete1** [comment]
hum what does 'git status' give?

**Comment by sahith22** [comment]
Upon using git status it gives-
You are currently cherry-picking commit ed6a47e.
  (all conflicts fixed: run "git cherry-pick --continue")
  (use "git cherry-pick --abort" to cancel the cherry-pick operation)

Changes to be committed:

	modified:   APMrover2/Parameters.cpp
	modified:   APMrover2/mode_loiter.cpp

**Comment by IamPete1** [comment]
sounds like you need to do git commit

some thing like 'git commit -a -m "Rover: Added reversing loiter_type parameter" '

I can recommend using github desktop, it makes it abit easier
